### ISSUE TITLE
How can I visualize the generated SQL code?

### ISSUE THREAD

**Issue by MrDave1999** [linq2db/linq2db.EntityFrameworkCore#247]
How can I visualize the generated SQL code? It is not displayed in the Visual Studio output box.

**Comment by MaceWindu** [comment]
We use EF.Core logging insfrastructure, so you just need to configure ef.core logging https://docs.microsoft.com/en-us/ef/core/logging-events-diagnostics/

**Comment by MaceWindu** [comment]
Closing as answered

**Comment by MrDave1999** [comment]
@MaceWindu Sorry for the late reply. I have not been able to solve the problem.
EF Core is supposed to fully integrate with `Microsoft.Extensions.Logging` when I call the `AddDbContext` method but still does not work when I call linq2db's `UpdateAsync` method.
The sql code is not displayed.

**Comment by MaceWindu** [comment]
@MrDave1999 you probably need to enable logging on linq2db side, check [this](https://github.com/linq2db/linq2db.EntityFrameworkCore/issues/38#issuecomment-691538799). Otherwise provide your logging configuration code so we can examine what is missing.

**Comment by AntonC9018** [comment]
> @MrDave1999 you probably need to enable logging on linq2db side, check [this](https://github.com/linq2db/linq2db.EntityFrameworkCore/issues/38#issuecomment-691538799). Otherwise provide your logging configuration code so we can examine what is missing.

Regular EF Core queries log it no problem. I'm trying a Merge query, and it doesn't log. Even with the trace switch turned on.

**Comment by AntonC9018** [comment]
Specifying a log action works, but we both know that's not a solution.

**Comment by MaceWindu** [comment]
Yeah, something is not right with logging configuration and we should review it as too many people hit issue being unable to setup it properly.
Could you add information what was missing in your configuration?

**Comment by AntonC9018** [comment]
It's basically this:
```cs
var host = Host.CreateDefaultBuilder(args)
    .ConfigureServices((hostContext, services) =>
    {
        var connectionString = hostContext.Configuration
            .GetValue<string>("Configuration:MainDatabaseConnectionString");
        services.AddSqlServer<MyDbContext>(connectionString);
    })
    .ConfigureAppConfiguration(builder =>
    {
        builder.AddJsonFile("appsettings.json");
    })
    .Build();

var serviceProvider = host.Services;
using var scope = serviceProvider.CreateScope();

LinqToDBForEFTools.Implementation = LinqToDbEfCoreImplWithProjectableSupport.Instance;
LinqToDBForEFTools.Initialize();
DataConnection.TurnTraceSwitchOn();
DataConnection.WriteTraceLine = static (message, displayName, _) =>
{
    Console.WriteLine(message);
};
```
EF Core logs by default, so I expect the lib to do too.

**Comment by sdanyliv** [comment]
Extension supports `ILoggerFactory`. I have made logging from the beginning of `linq2db.EntityFrameworkCore`, starting from EF Core 1, which supported only this mechanism. Well, something needs to be improved.

Try this example while configuring your `DbContext`:
```cs
services.AddDbContext<SomeContext>(options => 
    options
        .UseSqlServer(Configuration.GetConnectionString("DefaultConnection"))
        .UseLoggerFactory(LoggerFactory.Create(builder => builder.AddDebug()))
        .EnableSensitiveDataLogging()
    );
```
Turning trace switch on is not required.

**Comment by AntonC9018** [comment]
Thanks
